tests/basic/test_one_shot_keys.cpp

```diff
@@ -277,6 +273,56 @@ TEST_F(OneShot, OSMHoldNotLockingOSMs) {
     VERIFY_AND_CLEAR(driver);
 }
 
+TEST_F(OneShot, OSMTapThenHoldShouldSendEmptyReport) {
+    TestDriver driver;
+    InSequence s;
+    KeymapKey  osm_key     = KeymapKey{0, 0, 0, OSM(MOD_LSFT), KC_LSFT};
+    KeymapKey  regular_key = KeymapKey{0, 1, 0, KC_A};
+
+    set_keymap({osm_key, regular_key});
+
+    /*
+     * for some reason, none of the OSM keys are registered
+     * */
+
+    /* Press and release OSM */
+    EXPECT_NO_REPORT(driver);
+    tap_key(osm_key);
+    EXPECT_EQ(get_oneshot_mods(), 0x02); // verify oneshots are turned on
+    VERIFY_AND_CLEAR(driver);
+
+    /* Press OSM */
+    EXPECT_NO_REPORT(driver);
+    osm_key.press();
+    run_one_scan_loop(); // not yet registered
+    VERIFY_AND_CLEAR(driver);
+
+    /* Hold OSM */
+    EXPECT_REPORT(driver, (osm_key.report_code)).Times(1); // report should be send now
+    idle_for(TAPPING_TERM);
+    EXPECT_EQ(get_oneshot_mods(), 0x00); // key registered as hold, OSM should be disabled
+    VERIFY_AND_CLEAR(driver);
```

Comment:
This is not triggering a keyboard report, even though a hold should simply trigger the normal modifier keydown event after the TAPPING_TERM. Any help is more than welcome!

Reply:
Thank you for your efforts! Something that could help investigation is to turn on [debug logging](https://docs.qmk.fm/faq_debug#debugging) and turn on verbose action debug logging by adding to config.h:

```.c
#define ACTION_DEBUG
```

It then *should* happen that holding an OSM key logs the message "`MODS_TAP: Oneshot: 0`," corresponding to this line: 

https://github.com/qmk/qmk_firmware/blob/master/quantum/action.c#L467

And if not, the logged messages will hopefully point to what's going wrong.